api/apps/api/src/modules/cost-surface/cost-surface.service.ts

```diff
@@ -52,4 +60,47 @@ export class CostSurfaceService {
     const costSurface = await this.costSurfaceRepository.save(instance);
     return right(costSurface);
   }
+
+  async updateCostSurfaceShapefile(
+    userId: string,
+    projectId: string,
+    costSurfaceId: string,
+    dto: UpdateCostSurfaceDto,
+  ): Promise<
+    Either<
+      | typeof projectNotEditable
+      | typeof costSurfaceNotFoundForProject
+      | typeof costSurfaceNameAlreadyExistsForProject,
+      CostSurface
+    >
+  > {
+    if (
+      !(await this.projectAclService.canEditCostSurfaceInProject(
+        userId,
+        projectId,
+      ))
+    ) {
+      return left(projectNotEditable);
+    }
+
+    const nameCount = await this.costSurfaceRepository.count({
+      where: { name: dto.name },
+    });
+    if (nameCount > 0) {
+      return left(costSurfaceNameAlreadyExistsForProject);
```

Comment:
if we throw `costSurfaceNameAlreadyExistsForProject` shouldn't we filter by project as well?
plus, maybe for being consistent through the code, I would use `find()` and then check if the result is truthy or not, but just my two cents

Reply:
You're right, completely missed this. Thanks.